google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GrpcSpannerRpc.java

```diff
@@ -97,25 +100,23 @@
 public class GrpcSpannerRpc implements SpannerRpc {
   private static final Logger logger = Logger.getLogger(GrpcSpannerRpc.class.getName());
 
-  private static final Metadata.Key<String> API_CLIENT_KEY =
-      Metadata.Key.of("x-goog-api-client", Metadata.ASCII_STRING_MARSHALLER);
-  private static final Metadata.Key<String> RESOURCE_PREFIX_KEY =
-      Metadata.Key.of("google-cloud-resource-prefix", Metadata.ASCII_STRING_MARSHALLER);
   private static final Pattern DATABASE_PATTERN =
-      Pattern.compile("^(?<database>projects/[^/]*/instances/[^/]*/databases/[^/]*)(.*)?");
+      Pattern.compile("^(?<headerValue>projects/[^/]*/instances/[^/]*/databases/[^/]*)(.*)?");
   private static final Pattern INSTANCE_PATTERN =
-      Pattern.compile("^(?<instance>projects/[^/]*/instances/[^/]*)(.*)?");
+      Pattern.compile("^(?<headerValue>projects/[^/]*/instances/[^/]*)(.*)?");
```

Comment:
why not just define these two in SpannerHeaderProvider and use there instead of passing from here?

Reply:
This is possible. I was considering this too. Moved to SpannerHeaderProvider